app/src/main/java/ca/wheresthebus/adapter/StopSuggestionAdapter.kt

```diff
@@ -0,0 +1,81 @@
+package ca.wheresthebus.adapter
+
+import android.app.AlertDialog
+import android.os.Build
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
+import androidx.annotation.RequiresApi
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.RecyclerView
+import ca.wheresthebus.MainDBViewModel
+import ca.wheresthebus.R
+import ca.wheresthebus.data.ModelFactory
+import ca.wheresthebus.data.model.BusStop
+import ca.wheresthebus.data.model.FavouriteStop
+
+class StopSuggestionAdapter(
+    private val suggestedStops: ArrayList<BusStop>,
+    private val mainDBViewModel: MainDBViewModel,
+    private val modelFactory: ModelFactory
+) : RecyclerView.Adapter<StopSuggestionAdapter.SuggestedStopViewHolder>() {
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    inner class SuggestedStopViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        private val stopName: TextView = view.findViewById(R.id.suggested_stop_name)
+        private val stopCode: TextView = view.findViewById(R.id.suggested_stop_code)
+        private val routesAtStop: TextView = view.findViewById(R.id.suggested_stop_routes)
+
+        fun bind(stop: BusStop) {
+            stopName.text = stop.name
+            stopCode.text = stop.code.value
+            val routeShortNames = ArrayList<String>()
+            for (route in stop.routes) {
+                routeShortNames.add(route.shortName)
+            }
+            routesAtStop.text = routeShortNames.joinToString(", ")
+        }
+
+        init {
+            view.setOnClickListener {
+                val selectedStop = suggestedStops[adapterPosition]
+                val builder = AlertDialog.Builder(view.context)
+                val customView = LayoutInflater.from(view.context).inflate(R.layout.dialog_add_fav_stop, null)
+                builder.setView(customView)
+                val nicknameEditText: EditText = customView.findViewById(R.id.nicknameEditText)
+                val routeEditText: EditText = customView.findViewById(R.id.routeEditTextField)
```

Comment:
Instead of a field, can this be a spinner or something that has all the routes that go to this stop? Otherwise it's hard to know what routes are available.

Reply:
👍